lib/yrewrite/seo.php

```diff
@@ -91,7 +91,9 @@ public function getTags(): string
         $tags['robots'] = '<meta name="robots" content="'.$content.'">';
 
         $canonicalUrl = rex_escape($this->getCanonicalUrl());
-        $tags['canonical'] = '<link rel="canonical" href="'.$canonicalUrl.'" />';
+        if (1 == $this->article->getValue(self::$meta_index_field) || (0 == $this->article->getValue(self::$meta_index_field) && $this->article->isOnline())) {
```

Comment:
Wieso bei noindex und online trotzdem der canonical-tag?

Reply:
Ich vermute Mal, dass der Artikel in einer anderen Sprache ja dennoch indexiert werden darf.

Reply:
Der canonical Tag soll ja nur angezeigt werden, wenn Indexierung erlaubt ist. Wenn in dem Auswahlfeld der Wert 1 ausgewählt ist, bedeutet das "Indexierung erlauben / in Sitemap anzeigen (Index)". Bei 0 "In Abhängigkeit von Status Indexierung erlauben und in Sitemap anzeigen".  Beides wird geprüft. Die gleiche Prüfung erfolgt ein paar Zeilen vorher (https://github.com/TobiasKrais/redaxo_yrewrite/blob/73ab68e18a05f79e90b36d56cae7fe4dcb2cc98d/lib/yrewrite/seo.php#L88) wo der Wert für das robot Tag gesetzt wird. Nach meinen Tests funktioniert alles korrekt. Noindex und online wäre der Wert -1.

Reply:
Danke für die Erläuterung! Mir war nicht präsent, wie der Wert 0 beschrieben/definiert ist.

Reply:
Kein Problem. Lieber einmal mehr schreiben.